internal/sbi/producer/ue_context_management.go

```diff
@@ -217,6 +217,22 @@ func RegistrationAmf3gppAccessProcedure(registerRequest models.Amf3GppAccessRegi
 		ue, _ := udm_context.Getself().UdmUeFindBySupi(ueID)
 		oldAmf3GppAccessRegContext = ue.Amf3GppAccessRegistration
 	}
+	// TS 23.502 4.2.2.2.2 14d: UDM initiate a Nudm_UECM_DeregistrationNotification to the old AMF
+	// corresponding to the same (e.g. 3GPP) access, if one exists
+	if oldAmf3GppAccessRegContext != nil {
```

Comment:
why move dereg procedure here? `oldAmf3GppAccessRegContext` was already retrieved and will not be overwritten by new context.

Reply:
@tim-ywliu @YouShengLiu

I think we can create the goroutine for sending deregistration notifications.

Reply:
Yes, see https://github.com/free5gc/udm/pull/25#discussion_r1322547100